### ISSUE TITLE
Maybe this row has a bug which can cause a OOM error

### ISSUE THREAD

**Issue by mstar-kk** [sequelize/sequelize#13282]
https://github.com/sequelize/sequelize/blob/9f950cbcbdd659d559496b77c40e0f827b108561/lib/dialects/mysql/query.js#L122
If I use a bigInt data-type for id primary key, startId will be formatted to a string. In this situation using “+” to format id will cause OOM error.

**Comment by Spekpannenkoek** [comment]
When you do `a string` + `a number`, the result will be a string with the two values **concatenated**, not **added together**.

Let's say **5** rows were affected and `startId` is **4**. How many times will this loop iterate?

```
for (let loop = 0, i = startId; i < startId + data.affectedRows; i++) {
    console.log("iteration #", ++loop, 
      "value", i, "is a", typeof i, 
      ". startID + affectedRows =", startId + data.affectedRows);
}

/* Results in:
iteration # 1 value 4 is a string . startID + affectedRows = 45
iteration # 2 value 5 is a number . startID + affectedRows = 45
...
iteration # 40 value 43 is a number . startID + affectedRows = 45
iteration # 41 value 44 is a number . startID + affectedRows = 45
*/
```

You assume it will loop while `4 < 4+5`, which is 5 times. In fact, it loops while `"4" < "4"+5`, in other words, while `"4" < "45"`, which is 41 times.

When `startId` becomes longer, this becomes an even bigger problem. If `startId` is 400, this example would loop while `"400" < "400" + 5` = `"400" < "4005"` (3506 times!). Since you're using a big int for the id value, you are probably using much larger values than just 400 causing your loop to run out of control.

---

Luckily, if you use Node 10.4 or higher, there is a very easy solution.

[`BigInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt) is like the `Number` type except, as the name gives away, it supports 64 bit numbers.

There is a function named [`BigInt`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/BigInt/BigInt) that takes a number or a string. It has a `.toString` method should you need go the other way (`BigInt("4").toString() === "4"`). There's one catch: you can't do `BigInt(startId) + data.affectedRows`, because `data.affectedRows` is an integer:

> `Uncaught TypeError: Cannot mix BigInt and other types, use explicit conversions`

Easily fixed: Follow the given advice, do `BigInt(startId) + BigInt(data.affectedRows)`. 

---

```
for (let loop = 0, i = BigInt(startId); i < BigInt(startId) + BigInt(data.affectedRows); i++) {
    console.log("iteration #", ++loop, 
      "value", i, "is a", typeof i, 
      ". startID + affectedRows =", startId + data.affectedRows);
}

/* Results in:
iteration # 1 value 4n is a bigint . startID + affectedRows = 45
iteration # 2 value 5n is a bigint . startID + affectedRows = 45
iteration # 3 value 6n is a bigint . startID + affectedRows = 45
iteration # 4 value 7n is a bigint . startID + affectedRows = 45
iteration # 5 value 8n is a bigint . startID + affectedRows = 45
*/
```

Note how it logs `4n` (the 'n' indicates that it's a BigInt, not a Number) and that there's just 5 iterations. 

When (database) BigInt support was added, (JavaScript) BigInt support wasn't widespread (or perhaps even around?), which is why Sequelize uses strings.

Long story short:

- Not a bug in Sequelize, your loop just doesn't do what you expect it to. 
- Cast your (database) big ints (which are JavaScript strings) to (JavaScript) BigInts when comparing them.
